src/components/ImportPage/index.js

```diff
@@ -148,24 +222,129 @@ export default ({ oha, onChangeOHA, isDesktop }) => {
+      }
+    }
+    return taskOutput
+  }
+  function getSampleNameFromURL(sample) {
+    var sampleName
+    if (typeof sample.imageUrl !== "undefined") {
+      sampleName = sample.imageUrl.match(
+        `.*\\/(([^\\/\\\\&\\?]*)\\.([a-zA-Z0-9]*))(\\?|$)`
+      )
+    } else {
+      sampleName = sample.videoUrl.match(
+        `.*\\/(([^\\/\\\\&\\?]*)\\.([a-zA-Z0-9]*))(\\?|$)`
+      )
+    }
+    return sampleName
+  }
+  function searchSampleName(sampleName, myArray) {
+    var nameToSearch
+    for (var i = 0; i < myArray.length; i++) {
+      nameToSearch = getSampleNameFromURL(myArray[i])
+      if (typeof myArray[i].sampleName !== "undefined") {
+        nameToSearch[1] = myArray[i].sampleName
+      }
+      if (
+        nameToSearch[0] !== sampleName[0] &&
+        nameToSearch[1] === sampleName[1]
+      ) {
+        return true
+      }
+    }
+    return false
+  }
+
+  function giveSampleName(appendedTaskData) {
+    for (var i = 0; i < appendedTaskData.length; i++) {
+      var sampleName = getSampleNameFromURL(appendedTaskData[i])
+      var boolName = true
+      var v = 1
+      while (boolName) {
+        if (
+          searchSampleName(sampleName, oha.taskData) ||
+          searchSampleName(sampleName, appendedTaskData)
+        ) {
+          sampleName[1] = sampleName[2] + v.toString() + "." + sampleName[3]
+          v++
+        } else {
+          appendedTaskData[i].sampleName = sampleName[1]
+          boolName = false
+        }
+      }
+      appendedTaskData[i].sampleName = sampleName[1]
+    }
+    return appendedTaskData
+  }
+
   const closeDialog = () => changeDialog(null)
   const onAddSamples = useEventCallback(
-    (appendedTaskData, appendedTaskOutput) => {
-      let newOHA = setIn(
-        oha,
-        ["taskData"],
-        (oha.taskData || []).concat(appendedTaskData)
+    async (appendedTaskData, appendedTaskOutput, json, configImport) => {
+      var newOHA = {}
+      newOHA = setIn(
+        newOHA,
+        ["taskOutput"],
+        setAnnotations(newOHA.taskOutput, appendedTaskOutput, configImport)
       )
-      if (appendedTaskOutput) {
+      if (
+        json !== null &&
+        typeof json !== "undefined" &&
+        typeof json.content !== "undefined" &&
+        typeof json.fileName !== "undefined"
+      ) {
+        json.content.taskData = giveSampleName(json.content.taskData)
         newOHA = setIn(
           newOHA,
-          ["taskOutput"],
-          extendWithNull(oha.taskOutput || [], oha.taskData.length).concat(
-            appendedTaskOutput
-          )
+          ["taskData"],
+          (oha.taskData || []).concat(json.content.taskData)
+        )
+        newOHA = setIn(newOHA, ["interface"], json.content.interface)
+        if (typeof file.fileName === "undefined" || file.fileName === "unnamed")
+          file = setIn(file, ["fileName"], json.fileName)
+        file = setIn(file, ["content"], newOHA)
+        onChangeFile(file, true)
```

Comment:
gah we need `onChangeFile` just to change the file name :|

I don't see a way around this for now.

Reply:
Okay if you have a better idea feel free to ask. I will change it